File: ruby/grains/grains.rb
class Grains

  def self.square(number)
    if number == 1
      1
    else
      square(number - 1) * 2
    end
  end

  def self.total
    0.upto(64).reduce do |grains, per_square|
      grains + square(per_square)
    end
  end
end

File: python/grains/grains.py
def on_square(square):
    if square == 1:
        return 1
    else:
        return on_square(square - 1) * 2

def total_after(square):
    if square == 1:
        return 1
    else:
        grains = 0
        for n in range(1, square + 1):
            grains += on_square(n)
        return grains
